packages/contracts/src/dollar/BondingCurve.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import "./BancorFormula.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "./interfaces/IERC20Ubiquity.sol";
+// import "./interfaces/IERC1155Ubiquity.sol";
+import "./core/UbiquityDollarManager.sol";
+import "./core/UbiquityDollarToken.sol";
+import "../ubiquistick/interfaces/IUbiquiStick.sol";
+
+/**
+ * @title Bonding Curve
+ * @dev Bonding curve contract based on Bancor formula
+ * Inspired from Bancor protocol
+ * https://github.com/bancorprotocol/contracts
+ */
+contract BondingCurve is BancorFormula, Pausable {
+    uint256 constant ACCURACY = 10e18;
+
+    /// @dev Token issued by the bonding curve
+    // IUbiquiStick immutable token;
+    address public tokenAddr;
+    // IERC1155Ubiquity public token;
+    IUbiquiStick immutable token;
+
+    /// @dev Token used as collateral for minting the Token issued by the bonding curve
+    // IERC20Ubiquity immutable collateral;
+    address public collateral;
+
+    /// @dev Treasury address
+    address public treasuryAddress;
+
+    /// @dev The ratio of how much collateral "backs" the total Token
+    uint32 public connectorWeight;
+
+    /// @dev The intersecting price to mint or burn a Token when supply == PRECISION
+    uint256 public baseY;
+
+    /// @dev token id
+    uint256 public constant BONDING_TOKEN_ID = 1;
+
+    /**
+     * @dev Available balance of reserve token in contract
+     */
+    uint256 public poolBalance = 0;
+
+    /// @dev Current number of tokens minted
+    uint256 public tokenIds = 0;
+
+    /// @dev Mapping of tokens minted to address
+    mapping (address => uint256) public share;
+
+    UbiquityDollarManager public manager;
+
+    event Deposit(address indexed user, uint256 amount);
+
+    event Withdraw(uint256 amount);
+
+    modifier onlyBondingMinter() {
+        require(
+            manager.hasRole(manager.BONDING_MINTER_ROLE(), msg.sender), "not minter"
+        );
+        _;
+    }
+
+    modifier onlyPauser() {
+        require(
+            manager.hasRole(manager.PAUSER_ROLE(), msg.sender), "not pauser"
+        );
+        _;
+    }
+
+    modifier onlyParamsSet() {
+        require(
+            connectorWeight != 0 && baseY != 0, "not set"
+        );
+        _;
+    }
+
+    constructor(
+        address _manager,
+        address _token
+    ) {
+        require(_token != address(0), "NFT address empty");
+        tokenAddr = _token;
+        token = IUbiquiStick(_token);
+
+        manager = UbiquityDollarManager(_manager);
```

Comment:
here you should replace by IAccessControl cos we don't have it anymore

Reply:
Is this a new update?

Reply:
@zgorizzo69 Since we are moving to diamond, can I just use the ManagerFacet from the Diamond all together?

Reply:
BondingCurve as a facet makes sense !

Reply:
I will have it ready by tomorrow